### PR TITLE
Some error handling.

### PR DESCRIPTION
**Pull request by bbfsdev**

### CODE REVIEW

File: src/apps/AdminApp/AdminRoot.js [hunk 1]
```diff
@@ -507,8 +513,7 @@ class AdminRoot extends Component {
         const toAck = [feed_user.id];
 
         if(command_type === "audio-out") {
-            gdm.send(cmd, toAck, (cmd) => gateway.sendCmdMessage(cmd)).
-            then(() => {
+            gdm.send(cmd, toAck, (cmd) => gateway.sendCmdMessage(cmd)).then(() => {
```
Comment (edoshor) [inline src/apps/AdminApp/AdminRoot.js:516]: code style: each promise chain (`then`, `catch` or `finally`) should be on its own line.
Please fix throughout this PR

File: src/apps/AdminApp/AdminRoot.js [hunk 2]
```diff
@@ -528,7 +533,8 @@ class AdminRoot extends Component {
         this.sendCommandMessage(command_type);
         return;
 
-        const {gateways, feed_user, current_janus, current_room, command_status, gdm} = this.state;
+        /* TEMPORARY NOT USED - DELETE IF NOT NEEDED */
```
Comment (edoshor) [inline src/apps/AdminApp/AdminRoot.js:536]: remove the "delete" part.
Premoderation is a feature yet to have been used. 
ok to comment out code but no need to promote it's ultimate removal as of now

File: src/apps/AdminApp/AdminRoot.js [hunk 3]
```diff
@@ -812,23 +813,24 @@ class AdminRoot extends Component {
...
-              return (
-                  <Table.Row active={feed.id === this.state.feed_id} key={i} onClick={() => this.getUserInfo(feed_user)} >
-                      <Table.Cell width={10}>{qt ? q : ""}{feed.display.display}</Table.Cell>
-                      {/*<Table.Cell positive={st} width={1}>{st ? v : ""}</Table.Cell>*/}
-                      <Table.Cell width={1}></Table.Cell>
-                  </Table.Row>
-              )
+      const users_grid = feeds.map((feed,i) => {
+          if (!feed) {
+            return undefined;
```
Comment (edoshor) [inline src/apps/AdminApp/AdminRoot.js:818]: undefined is mostly used by js runtime. we at userland use null.

File: src/apps/AdminApp/AdminRoot.js [hunk 4]
```diff
@@ -996,15 +998,14 @@ class AdminRoot extends Component {
...
-                               gdm={this.state.gdm}
-                               onChatRoomsInitialized={this.onChatRoomsInitialized}/>
+                          <ChatBox user={user}
+                                   rooms={rooms}
+                                   selected_janus={current_janus}
+                                   selected_room={current_room}
+                                   selected_group={current_group}
+                                   selected_user={feed_user}
+                                   gateways={gateways}
+                                   gdm={this.state.gdm} />
```
Comment (edoshor) [inline src/apps/AdminApp/AdminRoot.js:1008]: please don't remove indication for chat room initialization. It should be reflected to the user.

File: src/apps/SDIOutApp/SDIOutApp.js [hunk 5]
```diff
@@ -96,27 +102,27 @@ class SDIOutApp extends Component {
...
-                this.postInitGateway(user, gateway);
-            })
-            .catch(err => {
-                console.error("[Shidur] error initializing gateway", gateway.name, err);
-                setTimeout(() => {
-                    this.initGateway(user, gateway);
-                }, 5000);
-            });
+            .then(() => this.postInitGateway(user, gateway)).catch(err => {
+              console.error("[Shidur] error initializing gateway. Retru in 5 sec.", gateway.name, err);
```
Comment (edoshor) [inline src/apps/SDIOutApp/SDIOutApp.js:112]: typo: Retru

File: src/apps/SDIOutApp/SDIOutApp.js [hunk 6]
```diff
@@ -96,27 +102,27 @@ class SDIOutApp extends Component {
...
+              captureException(err, {source: 'SDIOut', gateway: gateway?.name || 'undefined'});
+              setTimeout(() => {
+                  this.initGateway(user, gateway).catch(err =>
+                      captureException(err, {source: 'SDIOut', gateway: gateway?.name || 'undefined'}));
```
Comment (edoshor) [inline src/apps/SDIOutApp/SDIOutApp.js:116]: worth adding `console.error` here

File: src/apps/AdminApp/AdminRoot.js [hunk 7]
```diff
@@ -125,34 +124,41 @@ class AdminRoot extends Component {
...
             });
     };
 
     initGateways = (user) => {
         const gateways = GxyJanus.makeGateways("rooms");
         this.setState({gateways});
 
-        return Promise.all(Object.values(gateways).map(gateway => (this.initGateway(user, gateway))))
+        const gatewayToInitPromise = (gateway) => this.initGateway(user, gateway).catch(error => {
+          captureException(error, {source: 'AdminRoot', gateway: gateway?.name || 'undefined'});
```
[the remark this answers is not in the document]
Comment (bbfsdev) [reply src/apps/AdminApp/AdminRoot.js:144]: This is not true. If you get broken data from API, e.g., someone did some manual change to db and the name of the gateway is an empty string.

How do you prefer handling this?

File: src/apps/AdminApp/AdminRoot.js (hunk 2 above)
Comment (bbfsdev) [reply src/apps/AdminApp/AdminRoot.js:536]: Done.

File: src/apps/AdminApp/AdminRoot.js (hunk 3 above)
Comment (bbfsdev) [reply src/apps/AdminApp/AdminRoot.js:818]: Done.

File: src/apps/SDIOutApp/SDIOutApp.js (hunk 6 above)
Comment (bbfsdev) [reply src/apps/SDIOutApp/SDIOutApp.js:116]: Done.

File: src/apps/AdminApp/AdminRoot.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (bbfsdev) [reply src/apps/AdminApp/AdminRoot.js]: Done.

File: src/apps/AdminApp/AdminRoot.js (hunk 2 above)
Comment (bbfsdev) [reply src/apps/AdminApp/AdminRoot.js]: Done.

File: src/apps/AdminApp/AdminRoot.js (hunk 4 above)
Comment (bbfsdev) [reply src/apps/AdminApp/AdminRoot.js]: Added indicator for status in AdminRoot.

File: src/apps/ShidurApp/UsersQuad.js [hunk 8]
```diff
@@ -251,15 +251,15 @@ class UsersQuad extends Component {
         this.setState({fullscr: true, index: i, question: is_qst});
         //this.sdiGuaranteeAction("fullscr_group" , true, i, group, is_qst, [AUDIOOUT_ID, SDIOUT_ID, SNDMAN_ID]);
         this.sdiAction("fullscr_group" , true, i, group, is_qst);
-        if(is_qst) this.micMute(true, room, janus, i);
+        if(is_qst) this.props.fetchRooms().then(() => this.micMute(true, room, janus, i));
```
Comment (amnonbb) [inline src/apps/ShidurApp/UsersQuad.js:254]: What changed herer?

File: src/apps/ShidurApp/UsersQuad.js [hunk 9]
```diff
@@ -251,15 +251,15 @@ class UsersQuad extends Component {
...
     };
 
     toFourGroup = (i, group, cb , is_qst) => {
         let {room,janus} = group;
         console.log("[Shidur]:: Back to four: ");
         //this.sdiGuaranteeAction("fullscr_group" , false, i, group, is_qst, [AUDIOOUT_ID, SDIOUT_ID, SNDMAN_ID]);
         this.sdiAction("fullscr_group" , false, i, group, is_qst);
-        if(is_qst) this.micMute(false, room, janus, i);
+        if(is_qst) this.props.fetchRooms().then(() => this.micMute(false, room, janus, i));
```
Comment (amnonbb) [inline src/apps/ShidurApp/UsersQuad.js:262]: What changed herer?

**Pull request merged**